packages/ui/src/Common.ts

```diff
@@ -525,6 +530,10 @@ export interface BoxProps {
   testID?: string;
 }
 
+// If onClick is provided, add accessibility props.
+export type BoxProps = BoxPropsBase &
+  (BoxPropsBase extends {onClick: () => void} ? AccessibilityProps : {});
+
```

Comment:
We could also do something like this: 

`type BoxProps = | BoxPropsBase & { onClick?: undefined } | BoxPropsBase & { onClick: () => void } & AccessibilityProps;`

The way it is now is also great, but if we end up with additional props I think this would be cleaner as a general pattern to use.